### PR TITLE
Added timeout feature and support to update a picklist in record deta…

### PR DESCRIPTION
**Pull request by cs-utkarsh-surana**
Added timeout feature to change the message and support to update a picklist in the record details page after the specified time is elapsed.

UTC:

widget installation verified.
Checked the new feature working in FortiSOAR.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -18,10 +18,10 @@ An example of using the custom picklist message widget would be displaying a cus
 
 ## Release Notes
 
-Following enhancements have been made to the custom picklist message widget in version 1.1.0:
+Following enhancements have been made to the custom picklist message widget in version 2.0.0:
 
-- Added an option of displaying a spinner on the detail view of the record for the duration of completion of the task mentioned in the custom widget.
-- Added an option of displaying an icon associated with the custom message on the detail view of the record. 
+- Added an option to add timer to change the custom message.
```
Comment (cs-vikas-shelar) [inline README.md:23]: Verify readme content from Namrata Mam once.

File: widget/edit.controller.js [hunk 2]
```diff
@@ -2,23 +2,25 @@
 (function () {
     angular
         .module('cybersponse')
-        .controller('editCustomPicklistMessage110Ctrl', editCustomPicklistMessage110Ctrl);
+        .controller('editCustomPicklistMessageDev200Ctrl', editCustomPicklistMessageDev200Ctrl);
```
Comment (cs-vikas-shelar) [inline widget/edit.controller.js:5]: Remove 'Dev' from controller names in all places.

File: widget/edit.controller.js [hunk 3]
```diff
@@ -2,23 +2,25 @@
...
+        $scope.config.isTimeout = config.isTimeout || false
         $scope.config.picklits_mapping = $scope.config.picklits_mapping || {
         options: []
        };
-      
+       
         function _init() {
             if ($state.params.module) {
                 loadAttributes();
+                loadAttributesToUpdate();
```
Comment (cs-vikas-shelar) [inline widget/edit.controller.js:23]: Do we really need this method? As same can be processed as a part of loadAttributes() method. Here code redundancy can be avoided.

File: widget/edit.controller.js [hunk 4]
```diff
@@ -30,15 +32,26 @@
...
+            return item.name === $scope.config.picklistItem;
             });
             $scope.loadlistitem(selectedPicklistItem);
           }
-        }
-                                );
+        });
       }
+
+      $scope.loadListItemValuesToUpdate = function (picklistItem) {
```
Comment (cs-vikas-shelar) [inline widget/edit.controller.js:44]: The same code block is used from lines 65 to 72. That can be replaced with this function call.

File: widget/info.json [hunk 5]
```diff
@@ -15,7 +15,7 @@
             "https://repo.fortisoar.fortinet.com/fsr-widgets/customPicklistMessage-1.1.0/images/custom-picklist-message-view.png"
```
Comment (cs-vikas-shelar) [inline widget/info.json:15]: All 1.1.0-specific links here need to be updated with the latest version path.

File: widget/view.controller.js [hunk 6]
```diff
@@ -1,72 +1,113 @@
...
+         Modules.get({
+            module: $state.params.module,
+            id: $state.params.id,
+            __selectFields: $scope.config.picklistItem
+         }).$promise.then(function (result) {
+            var foundPickListValue = _.find($scope.config.picklits_mapping.options, function (item) {
+               return item.picklistValue === result[$scope.config.picklistItem].itemValue;
+            });
+            if (foundPickListValue) {
+               $scope.flag = true
```
Comment (cs-vikas-shelar) [inline widget/view.controller.js:24]: Will suggest giving some functionality-specific names here for the flag.

**Review by cs-vikas-shelar — CHANGES_REQUESTED** [review]
Done initial round of review. Few comments are added, please work on it.

File: README.md (hunk 1 above)
Comment (cs-suyog-jinturkar) [reply README.md:23]: @cs-utkarsh-surana , please add more details around this feature wrt user usability

File: README.md [hunk 7]
```diff
@@ -37,11 +37,16 @@ Following enhancements have been made to the custom picklist message widget in v
...
+| Show Icon                          | Select the **Show Icon** option if you want to display an icon associated with the custom message on the detail view of the record. Once you select the **Show Icon** option, then from the icons drop-down list, you can choose the icon to be displayed with the custom message. |
+| Do you want to configure a timeout | Specify whether you want to configure a timeout or not.                                                                                                                                                                                                                            |
+| Specify The Time In Minutes        | Specify the time in minutes.                                                                                                                                                                                                                                                       |
```
Comment (cs-suyog-jinturkar) [inline README.md:48]: Need more details in description about this option and how to use it

**Pull request closed without merging**